### PR TITLE
Closes #787: Migration Paragraph Single Card to AZ Cards

### LINKED ISSUE #787: Migrate Single Card paragraph type
See migration mapping doc for details: https://docs.google.com/spreadsheets/d/134UMiYtE8f1YAKYgSukL2JxFzDJF6-TgCAvYFC-vxqQ/edit#gid=0

### PR DESCRIPTION
**Pull request by smitahantzios**
Paragraph Single Card migration to az_paragraphs_cards

## Description
Paragraph Single Card migration to az_paragraphs_cards

## Related Issue

Closes #787 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by tadean — APPROVED** [review]

**Comment by camikazegreen** [comment]
I've attempted to rebuild in Probo to resolve the PHPUnit test failure, but it failed again: unable to connect to the database.

Anything else we need to look at for this, or just rebuild again?

**Review by tadean — DISMISSED** [review]

**Review by trackleft — DISMISSED** [review]

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - 0
+      - url
+
+  field_az_cards/body_format:  
+    plugin: default_value
+    default_value: az_standard
+    
+  behavior_settings:
+    plugin: paragraphs_behavior_settings
+    card_width: 'col-md-4 col-lg-4'
```
Comment (danahertzberg) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:52]:
```suggestion
    card_width: 'col-12'
```

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    card_style: 'card'
+    card_width_sm: 'col-sm-6'
```
Comment (danahertzberg) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:54]:
```suggestion
    card_width_sm: 'col-12'
```

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      - value
+
+  field_az_cards/link_title:
+    plugin: extract
+    source: field_uaqs_links
+    index:
+      - 0
+      - title
+
+  field_az_cards/link_uri:
```
Comment (danahertzberg) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:39]: Does it matter if line 39 says "uri" and line 44 says "url"?

**Review by danahertzberg — CHANGES_REQUESTED** [review]

**Comment by danahertzberg** [comment]
Where is the setting for card style to be bordered? Or is that not needed because it's the default?

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml (hunk 3 above)
Comment (smitahantzios) [reply modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:39]: Cards field is `link_uri` only.

**Comment by smitahantzios** [comment]
> Where is the setting for card style to be bordered and background white? Or is that not needed because it's the default?

@danahertzberg  - Below code in migration yml, will add the values by default.

```
behavior_settings:    plugin: paragraphs_behavior_settings    
card_width: 'col-12'    
card_style: 'card'    
card_width_sm: 'col-12'    
card_width_xs: 'col-12'
```

**Comment by tadean** [comment]
> > Where is the setting for card style to be bordered and background white? Or is that not needed because it's the default?
> 
> @danahertzberg - Below code in migration yml, will add the values by default.
> 
> ```
> behavior_settings:    
> plugin: paragraphs_behavior_settings    
> card_width: 'col-12'    
> card_style: 'card'    
> card_width_sm: 'col-12'    
> card_width_xs: 'col-12'
> ```

@smitahantzios Re: @danahertzberg's mention of white background, In addition to the paragraph behavior settings listed here, the individual card fields have a required subfield called `option` that contains **card**-specific settings. This is where the white background gets set.

The default value for this should be serialized php of `['class' => 'bg-white']`

See here for more information:  https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php#L164

**Comment by smitahantzios** [comment]
> > > Where is the setting for card style to be bordered and background white? Or is that not needed because it's the default?
> > 
> > 
> > @danahertzberg - Below code in migration yml, will add the values by default.
> > ```
> > behavior_settings:    
> > plugin: paragraphs_behavior_settings    
> > card_width: 'col-12'    
> > card_style: 'card'    
> > card_width_sm: 'col-12'    
> > card_width_xs: 'col-12'
> > ```
> 
> @smitahantzios Re: @danahertzberg's mention of white background, In addition to the paragraph behavior settings listed here, the individual card fields have a required subfield called `option` that contains **card**-specific settings. This is where the white background gets set.
> 
> The default value for this should be serialized php of `['class' => 'bg-white']`
> 
> See here for more information: https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php#L164

Done, added the card background.

**Comment by danahertzberg** [comment]
Need to add setting for `bordered cards`

**Review by tadean — APPROVED** [review]

**Review by danahertzberg — APPROVED** [review]
Looks good from a requirements perspective

**Review by trackleft — APPROVED** [review]

**Pull request merged**